remoteappmanager/rest/rest_handler.py

```diff
@@ -25,6 +26,44 @@ def get_resource_handler_or_404(self, collection_name):
         except KeyError:
             raise web.HTTPError(httpstatus.NOT_FOUND)
 
+    def write_error(self, status_code, **kwargs):
+        """Provides appropriate payload to the response in case of error.
+        """
+        exc_info = kwargs.get("exc_info")
+
+        if exc_info is None:
+            self.clear_header('Content-Type')
+            self.finish()
+
+        exc = exc_info[1]
+
+        if isinstance(exc, PayloadedHTTPError) and exc.payload is not None:
```

Comment:
This is part of the design decision. My vision is that this method only handles http-level exception types, with the conversion of exception type done at the individual http method handlers. This allows us to take appropriate action and, for example log contextually to the method handler. Converting here we lose this specificity. 

I don't know which one is best. For sure, once we were to go for a write_error method that handles rest exceptions, then it should handle all exceptions. Plus, write_error is just responsible for the payload. We would have to reimplement send_error too, so that proper status can be set.